src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -533,6 +541,46 @@ def _visit_where(self, node: foast.Call, **kwargs) -> foast.Call:
             location=node.location,
         )
 
+    def _extract_promoted_tuple_type(
+        self,
+        node_type_ls: list,
+        mask_type: ct.FieldType,
+        return_type: None | ct.FieldType | ct.ScalarType,
+    ) -> ct.FieldType:
+        for element in node_type_ls:
+            if isinstance(element, ct.TupleType):
+                return self._extract_promoted_tuple_type(element.types, mask_type, return_type)
```

Comment:
It seems a bit unintuitive that this function should return on the first `TupleType` element, completely discarding all succeeding elements. Is that intentional?

With the current nested tuple test it doesn't seem to make a difference either way, that's why I ask.